packages/sync/src/links/LocalDirectiveFilterLink.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  ApolloLink,
+  NextLink,
+  Observable,
+  Operation
+} from "apollo-link";
+import { hasDirectives, removeDirectivesFromDocument, checkDocument } from "apollo-utilities";
+import { LocalDirectives, MUTATION_QUEUE_LOGGER } from "../config/Constants";
+import debug from "debug";
+
+export const logger = debug(MUTATION_QUEUE_LOGGER);
+
+export class LocalDirectiveFilterLink extends ApolloLink {
+  private connectionRemoveOnlineOnly = {
+    name: "onlineOnly"
+  };
+  private connectionRemoveNoSquash = {
+    name: "noSquash"
+  };
+  constructor() {
+    super();
+  }
+  public request(operation: Operation, forward: NextLink) {
+    logger("Checking if client directives need to be removed on ", operation);
+    const clientDirectivesPresent = hasDirectives(
+      [
+        LocalDirectives.ONLINE_ONLY,
+        LocalDirectives.NO_SQUASH
+      ],
+      operation.query
+    );
+    if (!clientDirectivesPresent) {
+      return forward(operation);
+    } else {
+
+      checkDocument(operation.query);
```

Comment:
What this is for? I removed that and it still works as expected.

Reply:
Yeah, but any reason why we do this here? At this point document wasn't touched at all so we kinda duplicating validation that happened before?